### PR TITLE
Feature/ensure pods restart when configmap changes

### PR DESCRIPTION
**Pull request by yandongxiao**

### CODE REVIEW

File: helm-charts/charts/kube-starrocks/templates/_helpers.tpl [hunk 1]
```diff
@@ -120,3 +120,31 @@ be.conf: |
...
+  {{- printf "%s" $hash }}
+{{- end }}
+
+
+{{/*
+starrockscluster.be.config.hash is used to calculate the hash value of the be.conf, and due to the length limit, only
+the first 8 digits are taken, which will be used as the annotations for pods.
+*/}}
+{{- define "starrockscluster.be.config.hash" }}
+  {{- $hash := toJson .Values.starrocksBeSpec.config | sha256sum | trunc 8 }}
```
Comment (kevincai) [inline helm-charts/charts/kube-starrocks/templates/_helpers.tpl:139]: if only use the first 8 digits, how about change to crc32?

Comment (yandongxiao) [reply helm-charts/charts/kube-starrocks/templates/_helpers.tpl:139]: helm may not support crc32, see http://masterminds.github.io/sprig/

Comment (kevincai) [reply helm-charts/charts/kube-starrocks/templates/_helpers.tpl:139]: ok

File: helm-charts/charts/kube-starrocks/templates/_helpers.tpl [hunk 2]
```diff
@@ -120,3 +120,31 @@ be.conf: |
 {{- define "starrockscluster.be.log.path" -}}
 {{- print "/opt/starrocks/be/log" }}
 {{- end }}
+
+{{/*
+starrockscluster.fe.config.hash is used to calculate the hash value of the fe.conf, and due to the length limit, only
+the first 8 digits are taken, which will be used as the annotations for pods.
+*/}}
+{{- define "starrockscluster.fe.config.hash" }}
+  {{- $hash := toJson .Values.starrocksFESpec.config | sha256sum | trunc 8 }}
```
Comment (kevincai) [inline helm-charts/charts/kube-starrocks/templates/_helpers.tpl:129]: what if user provided an empty fe/be/cn config? will this be still ok?

**Review by kevincai — APPROVED** [review]

**Pull request merged**